src/controller/AutoCommissioner.cpp

```diff
@@ -240,6 +228,16 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStageInternal(Commissio
         // operational network because the provisioning of certificates will trigger the device to start operational advertising.
         if (mNeedsNetworkSetup)
         {
+            // if there is a WiFi or a Thread endpoint, then perform scan
```

Comment:
Why is this being moved to after SendNOC?

Reply:
Added a comment in code to explain this. 

// Perform Scan (kScanNetworks) and collect credentials (kNeedsNetworkCreds) right before configuring network.
// This order of steps allows the workflow to return to collect credentials again if network enablement fails.